File: C#-Server+Client/BaseLibrary/communication/command/handshake/ErrorCommand.cs
using System.Collections.Generic;
using BaseLibrary.communication.protocol;

namespace BaseLibrary.communication.command.handshake {
    public class ErrorCommand : AHandShakeCommand, ACommand.Sendable {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        public static readonly IFactory<ACommand.Sendable, ACommand> FACTORY = new CommandFactory();
        private sealed class CommandFactory : ACommandFactory {

            internal CommandFactory() : base() { }

            public override bool IsDeserializeable(string s) {
                if (s != null && s.StartsWith("ERROR")) {
                    var message = s.Substring(("ERROR").Length).Trim();
                    cache.Cached(s, new ErrorCommand(message));
                    return true;
                }
                return false;
            }

            public override bool IsTransferable(ACommand c) {
                if (c is ErrorCommand) {
                    var c1 = (ErrorCommand)c;
                    cache.Cached(c, new ErrorCommand(c1.MESSAGE));
                    return true;
                }
                return false;
            }
        }

        public string MESSAGE { get; private set; }

        public ErrorCommand(string message)
            : base() {
            MESSAGE = message;
        }

        public string Serialize() {
            return "ERROR "+ MESSAGE;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/handshake/ErrorCommand.java
package BaseLibrary.communication.command.handshake;

import BaseLibrary.communication.command.ACommand;
import BaseLibrary.communication.command.ACommandFactory;
import BaseLibrary.communication.protocol.AProtocol;
import BaseLibrary.communication.protocol.IFactory;

public final class ErrorCommand extends AHandShakeCommand {

	private static final String NAME = "ERROR";

	static {
		AProtocol.registerForDeserialize(NAME, ErrorCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	public static final IFactory<ACommand, ACommand> FACTORY = new CommandFactory();

	private static final class CommandFactory extends ACommandFactory {

		private CommandFactory() {
		}

		@Override
		public boolean isDeserializeable(String s, AProtocol protocol) {
			if (s != null && s.startsWith("ERROR")) {
				String message = s.substring(("ERROR").length()).trim();
				cache.cached(s, new ErrorCommand(message));
				return true;
			}
			return false;
		}

		@Override
		public boolean isTransferable(ACommand c) {
			if (c instanceof ErrorCommand) {
				ErrorCommand c1 = (ErrorCommand) c;
				cache.cached(c, new ErrorCommand(c1.MESSAGE));
				return true;
			}
			return false;
		}
	}

	public final String MESSAGE;

	public ErrorCommand(String message) {
		MESSAGE = message;
	}

	@Override
	public String serialize(AProtocol p) {
		return "ERROR " + MESSAGE;
	}
}
